### PR TITLE
Feature/11863 fss tooling implement set expiry date

### PR DESCRIPTION
**Pull request by rvijayavel**
This PR contains -
 Set expiry date implementation & related UTs
Added validations for expiry date in new batch job and UT
Moved macro into common place
Dotnet client backward compatibility related changes ( #12460)

### CODE REVIEW

File: FileShareService.DesktopClient.Core/Jobs/SetExpiryDateJob.cs [hunk 1]
```diff
@@ -1,27 +1,53 @@
...
+        public bool IsExpiryDateKeyExist { get; private set; }
+
         public void Validate(JToken jsonToken)
         {
-            //This validation will be implemented when PBI SetExpiryDateJob is picked-up.
+            #region Predeserialize validations
+            JToken? expiryDateToken = jsonToken.SelectToken("actionParams.expiryDate");
+
+            //Set value if key exists
+            IsExpiryDateKeyExist = expiryDateToken != null;
```
Comment (jimjamjahaa) [inline FileShareService.DesktopClient.Core/Jobs/SetExpiryDateJob.cs:24]: Rename variable from `IsExpiryDateKeyExist` to `ExpiryDateKeyExists` please

**Review by jimjamjahaa — CHANGES_REQUESTED** [review]
A very small thing, just a variable name change. 

Otherwise looks good.

File: FileShareService.DesktopClient.Core/Jobs/SetExpiryDateJob.cs (hunk 1 above)
Comment (rvijayavel) [reply FileShareService.DesktopClient.Core/Jobs/SetExpiryDateJob.cs]: Done rename to ExpiryDateKeyExists as suggested.

**Review by jimjamjahaa — APPROVED** [review]
lgtm

**Review by bhalcha13813 — APPROVED** [review]
looks good

**Review by crdandoukho — CHANGES_REQUESTED** [review]
Hi Vijay, 
Can you please check your changes have not resulted in less error information being reported to the user when a json file with invalid file structure is loaded into the app.
Please find attached screen captures showing the error message displayed for sampleActionsWithInvalidFormat.json in main and this branch. I think this branch is not displaying the error message correctly, could you please check?
I have also noticed in this branch the top error message is not cleared when reloading a different json file. Could you please address this too?
sampleActionsWithInvalidFormat_main
sampleActionsWithInvalidFormat_11860

**Comment by rvijayavel** [comment]
> Hi Vijay, Can you please check your changes have not resulted in less error information being reported to the user when a json file with invalid file structure is loaded into the app. Please find attached screen captures showing the error message displayed for sampleActionsWithInvalidFormat.json in main and this branch. I think this branch is not displaying the error message correctly, could you please check? I have also noticed in this branch the top error message is not cleared when reloading a different json file. Could you please address this too? sampleActionsWithInvalidFormat_main sampleActionsWithInvalidFormat_11860

@crdandoukho - This is missed from my side. Fixed it and modified the UT to check the error count also.

**Comment by rvijayavel** [comment]
@crdandoukho - This is missed from my side. Fixed it and modified the UT to check the error count also.

**Review by crdandoukho — APPROVED** [review]

**Pull request merged**